tst/CTA.Rules.Test/AwsRulesBaseTest.cs

```diff
@@ -139,22 +143,38 @@ public string GetSrcPath(string path)
                     CopyTestRules();
                     var analysisRunResult = solutionPort.AnalysisRun();
 
+                    StringBuilder str = new StringBuilder();
                     foreach (var projectResult in analysisRunResult.ProjectResults)
                     {
                         Assert.IsTrue(projectResult.ProjectActions.ToSummaryString()?.Length > 0);
+                        StringBuilder projectResults = new StringBuilder();
+                        projectResults.AppendLine(projectResult.ProjectFile);
+                        projectResults.AppendLine(projectResult.ProjectActions.ToString());
+                        result.ProjectResults.Where(p => p.CsProjectPath == projectResult.ProjectFile).FirstOrDefault().ProjectAnalysisResult = projectResults.ToString();
+
+                        str.Append(projectResults);
                     }
+                    result.SolutionAnalysisResult = str.ToString();
+
+
                     var runResult = solutionPort.Run();
-                    result = GenerateSolutionResult(solutionDir, analysisRunResult, runResult);
```

Comment:
Could you help me understand why we changed how we build the result. I'm not quite seeing how reading project paths from the solution file affects all this.

Reply:
Oh this was a side fix. I realized that we were iterating over the entire project list twice and re-doing a lot of the same work. So I remove this reference and decided to take advantage of current loops to fill in the return object instead, which sped up the UTs a bit also, but more importantly is also much more efficient.